v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/endtoend/EndToEndTestingITBase.java

```diff
@@ -362,4 +598,122 @@ protected JDBCSource createMySqlDatabase(
             Map.of(cloudSqlResourceManager.getDatabaseName(), tables.keySet().stream().toList()))
         .build();
   }
+
+  protected String generateSessionFile(
+      JDBCSource jdbcSourceShard,
+      CloudSqlResourceManager cloudSqlResourceManager,
+      SpannerResourceManager spannerResourceManager)
+      throws IOException, InterruptedException {
+    String spannerMigrationToolPath = System.getenv("spanner_migration_tool_path");
+    if (StringUtils.isBlank(spannerMigrationToolPath)) {
+      throw new RuntimeException(
+          "Error: spanner_migration_tool_path environment variable is not set or is empty.");
+    }
+    List<String> command = new ArrayList<>();
+    command.add(spannerMigrationToolPath);
+    command.add("schema");
+    command.add("--source=MySQL");
+    String sourceProfile =
+        String.format(
+            "host=%s,port=%s,user=%s,password=%s,dbName=%s",
+            jdbcSourceShard.hostname(),
+            jdbcSourceShard.port(),
+            jdbcSourceShard.username(),
+            jdbcSourceShard.password(),
+            cloudSqlResourceManager.getDatabaseName());
+    command.add("--source-profile=" + sourceProfile);
+    String targetProfile =
+        String.format("project=%s,instance=%s", PROJECT, spannerResourceManager.getInstanceId());
+    command.add("--target-profile=" + targetProfile);
+    command.add("--project=span-cloud-testing");
+
+    ProcessBuilder processBuilder = new ProcessBuilder(command);
+
+    Process process = processBuilder.start();
+    // Regex to capture the session filename
+    Pattern sessionFilePattern =
+        Pattern.compile("^Wrote session to file '([^']+\\.session\\.json)'\\.?$");
+
+    final List<String> capturedOutputLines = new ArrayList<>();
+    final List<String> capturedErrorLines = new ArrayList<>();
+    String[] tempCapturedSessionFileName = new String[1]; // Effectively final for lambda
+
+    ExecutorService executor = Executors.newFixedThreadPool(2);
+
+    // Read stdout
+    Future<?> stdoutFuture =
+        executor.submit(
+            () -> {
+              try (BufferedReader reader =
+                  new BufferedReader(new InputStreamReader(process.getInputStream()))) {
+                String line;
+                while ((line = reader.readLine()) != null) {
+                  System.out.println("TOOL_STDOUT: " + line); // Log all output
```

Comment:
Where do these system.out.println lines get logged? Will this be removed?

We should use LOG object for logging otherwise.